src/frontend/js/types/Joanie.ts

```diff
@@ -301,6 +309,27 @@ export interface NestedCredentialOrder extends AbstractNestedOrder {
 
 export type OrderEnrollment = Pick<Order, 'id' | 'state' | 'product_id' | 'certificate_id'>;
 
+export interface NestedCourseOrder {
+  id: Order['id'];
+  created_on: Order['created_on'];
+  owner: User;
+  course_id: Order['id'];
+  product_id: Order['id'];
+  state: Order['state'];
+  enrollment_id: Enrollment['id'];
+  organization: Organization;
+  certificate_id?: Order['certificate_id'];
+  product: DefinitionResourcesProduct;
+  contract: ContractLight;
+}
+
+export interface CourseOrderResourceQuery extends PaginatedResourceQuery {
+  course_id?: CourseListItem['id'];
```

Comment:
`course_id` looks required in the request logic

Reply:
yes but it did not work. Seems like our api typing refuse required parameters.
```ts
Type '{ get(filters?: CourseOrderResourceQuery | undefined): Promise<PaginatedResponse<NestedCourseOrder>>; }' does not satisfy the constraint 'ApiResourceInterface<NestedCourseOrder, ResourcesQuery>'.
  Types of property 'get' are incompatible.
    Type '(filters?: CourseOrderResourceQuery | undefined) => Promise<PaginatedResponse<NestedCourseOrder>>' is not assignable to type '(filters?: ResourcesQuery | undefined) => any'.
      Types of parameters 'filters' and 'filters' are incompatible.
        Type 'ResourcesQuery | undefined' is not assignable to type 'CourseOrderResourceQuery | undefined'.
          Property 'course_id' is missing in type 'ResourcesQuery' but required in type 'CourseOrderResourceQuery'.
```

Reply:
https://github.com/openfun/richie/pull/2282#discussion_r1499336334